### PR TITLE
feat(jans-lock): extend infrastructure for cedarling

### PR DESCRIPTION
**Pull request by sokorototo**
### Prepare

- [x] Read [PR guidelines](https://github.com/JanssenProject/jans/blob/main/docs/CONTRIBUTING.md#prs)
- [x] Read [license information](https://github.com/JanssenProject/jans/blob/main/LICENSE)

-------------------

#### Implementation Details
 - Correct DCR implementation
 - Cleaned up commit tree
 - Correct function signatures

-------------------
### Test and Document the changes
- [x] Static code analysis has been run locally and issues have been fixed
- [ ] Relevant unit and integration tests have been added/updated
- [ ] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

### CODE REVIEW

File: jans-lock/cedarling/config.toml [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
-remote_uri = "https://api.github.com/repos/nynymike/cedar-playground/contents/gluu_lock.cedarschema?ref=main"
-use_brotli_decompression = false # Whether the policy store is to be decopmressed using Brotli
+# whether Cedarling should listen for SSE config updates
+dynamic-configuration = false
+openid-config-url = "https://account.gluu.org/.well-known/openid-configuration"
+
+[policy-store]
+# How to get policy store, can be local, remote or authenticated
+strategy = "local"
+uri = "https://api.github.com/repos/nynymike/cedar-playground/contents/gluu_lock.cedarschema?ref=main"
```
Comment (SafinWasi) [inline jans-lock/cedarling/config.toml:15]: We need to eventually check this into the main Jans repo, and provide a static link to it.

File: jans-lock/cedarling/tests/fetch-open-id-config.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+use cedarling::open_id_config;
+use wasm_bindgen_test::*;
+
+#[wasm_bindgen_test]
+async fn test() {
+	let config = open_id_config("https://account.gluu.org/.well-known/openid-configuration").await;
+	console_log!("Fetched config: {:?}", &config);
```
Comment (SafinWasi) [inline jans-lock/cedarling/tests/fetch-open-id-config.rs:7]: Tests should be done through assertions, console logging isn't testing anything. Also, doing real HTTP requests during tests will be very performance intensive. We want to do request testing via mocking the responses in some way.

File: jans-lock/cedarling/tests/fetch-schema.rs [hunk 3]
```diff
@@ -1,8 +1,14 @@
-use cedarling::utils;
+use cedar_policy::*;
 use wasm_bindgen_test::*;
 
 #[wasm_bindgen_test]
 async fn test() {
-	let schema = utils::fetch_schema().await;
+	let schema = include_str!("../policy-store/schema.txt");
```
Comment (SafinWasi) [inline jans-lock/cedarling/tests/fetch-schema.rs]: This is an empty file, and is not testing anything.

File: jans-lock/cedarling/tests/fetch-schema.rs [hunk 4]
```diff
@@ -1,8 +1,14 @@
...
+	let (schema, warnings) = Schema::from_str_natural(schema).unwrap();
+
+	for warning in warnings {
+		console_log!("Schema Parser generated Warning: {}", warning);
+	}
+
 	console_log!("Fetched Schema: {:?}", &schema)
```
Comment (SafinWasi) [inline jans-lock/cedarling/tests/fetch-schema.rs]: Needs to test via assertions

File: jans-lock/cedarling/tests/get-str.rs [hunk 5]
```diff
@@ -1,10 +1,9 @@
-use cedarling::utils;
-use wasm_bindgen::UnwrapThrowExt;
+use cedarling::http::{self, ResponseEx};
 use wasm_bindgen_test::*;
 
 #[wasm_bindgen_test]
 async fn test() {
-	let data = utils::get("https://site-scraper.sokorototo.workers.dev/").await.unwrap_throw();
-	let string = std::str::from_utf8(&data).unwrap_throw();
+	let data = http::get("https://site-scraper.sokorototo.workers.dev/", &[]).await.unwrap();
```
Comment (SafinWasi) [inline jans-lock/cedarling/tests/get-str.rs:6]: Needs to be done via mocking responses, not real requests.

**Review by SafinWasi — CHANGES_REQUESTED** [review]
Two main concerns:
- the tests are performing real HTTP requests. This should be done via mocking the HTTP server and responses. 
- the tests themselves look incomplete, as they're not doing any comparison or assertions for the expected vs actual output.

File: jans-lock/cedarling/tests/fetch-open-id-config.rs (hunk 2 above)
Comment (sokorototo) [reply jans-lock/cedarling/tests/fetch-open-id-config.rs:7]: Oh yeah, I saw the output and moved on. I'll add some assertions

File: jans-lock/cedarling/tests/fetch-schema.rs (hunk 3 above)
Comment (sokorototo) [reply jans-lock/cedarling/tests/fetch-schema.rs]: Not anymore, I just pushed a commit that read the policy store as JSON

File: jans-lock/cedarling/tests/fetch-schema.rs (hunk 4 above)
Comment (sokorototo) [reply jans-lock/cedarling/tests/fetch-schema.rs]: Schema is parsed from the PolicyStore JSON, and that process doesn't emit any warnings sadly

File: jans-lock/cedarling/tests/get-str.rs (hunk 5 above)
Comment (sokorototo) [reply jans-lock/cedarling/tests/get-str.rs:6]: It's a personal service, does it still need to use mocking?

File: jans-lock/cedarling/config.toml (hunk 1 above)
Comment (sokorototo) [reply jans-lock/cedarling/config.toml:15]: I use a static TOML file to make development easier and reproducible, later on, we'll probably use compile-time environment variables.

File: jans-lock/cedarling/tests/get-str.rs (hunk 5 above)
Comment (SafinWasi) [reply jans-lock/cedarling/tests/get-str.rs:6]: Mm yeah, it will still be using traffic that might cost depending on where we run tests.

Comment (sokorototo) [reply jans-lock/cedarling/tests/get-str.rs:6]: I don't think it will hit the service 1M times a month. Then it would cost me 5$

**Review by SafinWasi — DISMISSED** [review]

**Review by moabu — APPROVED** [review]

**Review by moabu — APPROVED** [review]

**Pull request merged**